enterprise/server/remote_execution/containers/ociruntime/ociruntime_test.go

```diff
@@ -922,6 +940,128 @@ func TestOverlayfsEdgeCases(t *testing.T) {
 	assert.Equal(t, 0, res.ExitCode)
 }
 
+// uncompressedLayer implements partial.UncompressedLayer from raw bytes.
+type uncompressedLayer struct {
+	diffID    containerregistry.Hash
+	mediaType types.MediaType
+	content   []byte
+}
+
+// DiffID implements partial.UncompressedLayer
+func (ul *uncompressedLayer) DiffID() (containerregistry.Hash, error) {
+	return ul.diffID, nil
+}
+
+// Uncompressed implements partial.UncompressedLayer
+func (ul *uncompressedLayer) Uncompressed() (io.ReadCloser, error) {
+	return io.NopCloser(bytes.NewBuffer(ul.content)), nil
+}
+
+// MediaType returns the media type of the layer
+func (ul *uncompressedLayer) MediaType() (types.MediaType, error) {
+	return ul.mediaType, nil
+}
+
+var _ partial.UncompressedLayer = (*uncompressedLayer)(nil)
+
+func TestHighLayerCount(t *testing.T) {
+	// Load busybox oci image
+	busyboxPath, err := runfiles.Rlocation(ociBusyboxRlocationpath)
+	require.NoError(t, err)
+	idx, err := layout.ImageIndexFromPath(busyboxPath)
+	require.NoError(t, err)
+	m, err := idx.IndexManifest()
+	require.NoError(t, err)
+	require.Equal(t, 1, len(m.Manifests))
+	require.True(t, m.Manifests[0].MediaType.IsImage())
+	busyboxImg, err := idx.Image(m.Manifests[0].Digest)
+	require.NoError(t, err)
+
+	for _, tc := range []struct {
+		layerCount int
+	}{
+		{layerCount: 5},
+		{layerCount: 20},
+		{layerCount: 21},
+		{layerCount: 58},
+		{layerCount: 128},
+	} {
+		// Note that the "busybox" oci image has 1 layer
+		// and the following tests will add more layers on top of it.
+		t.Run(fmt.Sprintf("1And%dLayers", tc.layerCount), func(t *testing.T) {
+			// Create new layers on top of busybox
+			var lastContent string
+			var layers []containerregistry.Layer
+			for i := range tc.layerCount {
+				lastContent = fmt.Sprintf("layer %d", i)
+				content := []byte(lastContent)
+
+				var b bytes.Buffer
+				hasher := crypto.SHA256.New()
+				mw := io.MultiWriter(&b, hasher)
+
+				tw := tar.NewWriter(mw)
+				err := tw.WriteHeader(&tar.Header{
+					Name:     "a.txt",
+					Size:     int64(len(content)),
+					Typeflag: tar.TypeReg,
+				})
+				require.NoError(t, err)
+				_, err = tw.Write(content)
+				require.NoError(t, err)
+				err = tw.Close()
+				require.NoError(t, err)
+
+				layer, err := partial.UncompressedToLayer(&uncompressedLayer{
+					diffID: containerregistry.Hash{
+						Algorithm: "sha256",
+						Hex:       hex.EncodeToString(hasher.Sum(make([]byte, 0, hasher.Size()))),
+					},
+					mediaType: types.OCILayer,
+					content:   b.Bytes(),
+				})
+				require.NoError(t, err)
+				layers = append(layers, layer)
+			}
+			testImg, err := mutate.AppendLayers(busyboxImg, layers...)
+			require.NoError(t, err)
+
+			// Start registry and push our new image there
+			ts := httptest.NewServer(registry.New(registry.Logger(log.New(io.Discard, "", 0))))
```

Comment:
need a `defer ts.Close()` here?

Reply:
Good call. Initially I was playing around with our own `testhttp` package which use `t.Cleanup` automatically. Later on I switched to the stdlib `httptest` and forgot about `Close()` 🤦